core/src/main/java/org/neo4j/ogm/session/transaction/support/TransactionFunctionWithoutResult.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2002-2018 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This product is licensed to you under the Apache License, Version 2.0 (the "License").
+ * You may not use this product except in compliance with the License.
+ *
+ * This product may include a number of subcomponents with
+ * separate copyright notices and license terms. Your use of the source
+ * code for these subcomponents is subject to the terms and
+ *  conditions of the subcomponent's license, as noted in the LICENSE file.
+ */
+
+package org.neo4j.ogm.session.transaction.support;
+
+/**
+ * Callback to be executed inside an OGM managed transaction.
+ * Private API not intended to be used by end users.
+ * @param <T> The return type of the callback.
+ * @author Nicolas Mervaillie
+ */
+@FunctionalInterface
+public interface TransactionFunctionWithoutResult {
```

Comment:
This is horrible name :-) what about `TransactionUnit`?  or `TransactionMethod`? @meistermeier wdyt?

Reply:
You are right, it does not really sound sophisticated. In the time we discussed this, it felt ok. After thinking about this a little bit longer, I have come to the conclusion that it might be better to prefix the classes with `Transactional...` instead of `Transaction...`. Maybe also the generic name `...Function` implies just a mapping instead of a db query and result processing. But I have really no idea how to call it. I would go with `TransactionalUnit`.